application/client/src/app/ui/elements/recent.mini/state.ts

```diff
@@ -5,17 +5,19 @@ import { recent } from '@service/recent';
 import { Subject } from '@platform/env/subscription';
 import { IlcInterface } from '@service/ilc';
 import { ChangesDetector } from '@ui/env/extentions/changes';
+import { Holder } from '@module/matcher/holder';
 
 export type CloseHandler = () => void;
 
-export class State {
+export class State extends Holder {
     public filter: Filter;
     public actions: WrappedAction[] = [];
     public update: Subject<void> = new Subject<void>();
     public selected: string = '';
     protected close: CloseHandler | undefined;
 
     constructor(ilc: IlcInterface & ChangesDetector) {
+        super();
```

Comment:
BTW, if you need to destroy matcher as soon as state's parent component was killed, you can add this:
```
ilc.life().destroy(() => {
            this.matcher.destroy();
            // it will be called as soon as component would be destroyed.
});
```

Reply:
We don't have a dedicated function to destroy the matcher.
An alternative that comes to my mind would be to call `delete this.matcher` but that is not really necessary, because that's being done internally anyway.